service/history/mutableStateTaskGenerator.go

```diff
@@ -139,12 +139,16 @@ func (r *mutableStateTaskGeneratorImpl) generateWorkflowStartTasks(
 	executionInfo := r.mutableState.GetExecutionInfo()
 	startVersion := startEvent.GetVersion()
 
+	expirationTime := timestamp.TimeValue(executionInfo.RetryExpirationTime)
```

Comment:
Actually it is possible to have `RetryExpirationTime` to be infinite and still have a real workflow run timeout.  In this case we should still create the workflow run timer.  Current logic will just bailout on creating the timer.

Reply:
yep, plz review this PR #1025 first, then i will send out the updated PR #1023, then i will update this section.